src/opm/input/eclipse/Schedule/Well/Well.cpp

```diff
@@ -1300,6 +1312,60 @@ bool Well::handleWPIMULT(const DeckRecord& record) {
 }
 
 
+bool Well::handleWINJMULT(const Opm::DeckRecord& record, const KeywordLocation& location) {
+    // for this keyword, the default for I, J, K will be negative
+    // it is not totally clear how specifying 0 or a negative values will work
+    // current match_eq function only treats 0 and default values for all connections,
+    // we might need to revisit this part later when complication regarding this occurs.
+    // it is possible that changing (item.get<int>(0) == 0); to (item.get<int>(0) <= 0) is solution to go
+    // while it remains to be discussed.
+    auto match = [=] ( const Connection& c) -> bool {
+        if (!match_eq(c.getI()  , record, "I", -1)) return false;
+        if (!match_eq(c.getJ()  , record, "J", -1)) return false;
+        if (!match_eq(c.getK()  , record, "K", -1)) return false;
+
+        return true;
+    };
+
+    // check whether it was under WREV previously.
+    // if yes, we need to reset all the connections for INJMULT specification
+    const bool is_prev_wrev = this->inj_mult_mode == InjMultMode::WREV;
+    using Kw = ParserKeywords::WINJMULT;
+    const InjMultMode mode = injMultModeFromString(record.getItem<Kw::MODE>().getTrimmedString(0), location);
+    this->inj_mult_mode = mode;
```

Comment:
Oh, I didn't notice this when doing the initial review.  This will unconditionally assign `inj_mult_mode`, even in the case, albeit unlikely, that no connections match the selection.  Is that the appropriate semantics?

Reply:
With some testing output, it shows we will unconditionally assign `inj_mult_mode`, even the new record does not match any existing connections. 

But for that case, I updated the code a little bit ([3b731a3](https://github.com/OPM/opm-common/pull/3161/commits/3b731a3da29d4eb8f9267437f375bb5a6eeb83df)) to make sure the updated `inj_mult_mode` will be put into the snapshots.

Reply:
The reason behind that is because we do not store `inj_mult_mode` in the `Connection`. `updateConnections()` will return `false` because the connections stay unchanged.